Meta/Lagom/Fuzzers/FuzzLzmaDecompression.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2023, Tim Schumacher <timschumi@gmx.de>.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/MemoryStream.h>
+#include <LibCompress/Lzma.h>
+#include <stdio.h>
+
+extern "C" int LLVMFuzzerTestOneInput(uint8_t const* data, size_t size)
+{
+    // These settings have been taken from the a.lzma example in the LZMA specification.
+    // Fuzzing the settings is not currently viable as archives can request a dictionary size of up to 4GB.
```

Comment:
Can we fuzz all options *except* the dictionary size? i.e. overwrite the dictionary size option after reading the options from the provided data.

Reply:
We could do that, but that would result in files that look like LZMA archives (not that they have some kind of magic), but would have to be changed first to yield the same results as the fuzzer.

What we _could_ do is to define some arbitrary "largest reasonable dictionary size" and reject any test inputs that have a larger dictionary size than that. LibFuzzer has a dedicated return value to indicate "don't add this to the test corpus", which would be perfect for this.

Reply:
That sounds like the ideal solution to me then.